bin/kafka-run-class.sh

```diff
@@ -107,7 +107,7 @@ else
   for file in "$base_dir"/streams/upgrade-system-tests-$SHORT_VERSION_NO_DOTS/build/libs/kafka-streams-upgrade-system-tests*.jar;
   do
     if should_include_file "$file"; then
-      CLASSPATH="$CLASSPATH":"$file"
+      CLASSPATH="$file":"$CLASSPATH"
```

Comment:
Cannot follow. Can you elaborate? Also, why was this not an issue in the past?

Reply:
Hmm, now that I think about this more, if memory serves, this is not a reliable way to ensure the right class gets loaded, since the classloader spec specifically leaves undefined how a class should be chosen when multiple are available.

Maybe we should just give the classes unique names instead?

Reply:
That's true extension classes (https://docs.oracle.com/javase/8/docs/technotes/tools/findingclasses.html#extclass)  but for user classes, the first one found on the classpath is what is used

Reply:
>Cannot follow. Can you elaborate? Also, why was this not an issue in the past?

When searching for classes on the classpath, if multple classes with the same name exist on the classpath, the first one found wins.  This was causing the failures of the test, as it was using the `StreamsUpgradeTest` found in trunk.

For the `1.1.1` upgrade the test is using the `StreamsUpgradeTest` in trunk and not the `upgrade-system-tests-11` module and failing due changes how the test is loaded.  It wasn't an issue before because we were specifying those versions.

Reply:
Thanks for the reference. It might be a failure to read, but I couldn't find where it specifies that it searches the classpath in order. Only that it searches jar manifests in order.

It seems plausible that the file classloader does actually traverse the classpath in order, but it still looks like it's not guaranteed, since it's not in the spec.

It may not be a huge deal, since the only impact of loading the wrong class is a flaky test, which we can easily just fix again later on.